R/draw-dm.R

```diff
@@ -146,6 +146,47 @@ dm_get_all_columns <- function(x) {
 #'   dm_draw()
 #' @export
 dm_set_colors <- function(dm, ...) {
+  avail_cols <- dm_get_available_colors()$dm
+  quos <- enquos(...)
+  # If old fall-through syntax is being used:
+  if (!all(names(quos) %in% avail_cols) &&
+    all(names(quos) %in% src_tbls(dm))) {
+    warning(paste0(
+      "When setting colors with `dm_set_colors()`, ",
+      "the syntax is now as follows: `blue = table_name1, ",
+      "orange = table_name2` etc. (tidyselect supported) ",
+      "Forwarding the function call ",
+      "to the old implementation of `dm_set_colors()`."
+    ))
+    return(dm_set_colors2(dm, !!!quos))
+  }
+
+  avail_tables <- src_tbls(dm)
+  selected_tables <- tidyselect::vars_select(avail_tables, ...) %>%
+    # names will get integer suffixes if duplicated ("blue1", "blue2", ...)
```

Comment:
Do you have an example? I'd rather remove the `set_names()` call below.

Reply:
``` r
avail_tables <- c("t1", "t2")
tidyselect::vars_select(avail_tables, blue = dplyr::starts_with("t"))
#> blue1 blue2 
#>  "t1"  "t2"

# it works when only using 1-element variables:
tidyselect::vars_select(avail_tables, blue = t1, blue = t2)
#> blue blue 
#> "t1" "t2"

# it doesn't work when using multi-element vectors:
tidyselect::vars_select(avail_tables, blue = c(t1, t2))
#> blue1 blue2 
#>  "t1"  "t2"
```

<sup>Created on 2020-01-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

But I agree, that it's currently not a good solution, mainly if at some point we also allow hex color codes that may also contain digits. Not sure how to make fail-safe.

Reply:
The dev version of tidyselect has `eval_select()` with options to control naming, we should be able to use it.

Reply:
I just uploaded some changes: now it behaves differently depending on {tidyselect}-version. The only thing is, that in case someone has an old {tidyselect}-version, a NOTE will be produced by R CMD check:
```r
> checking dependencies in R code ... NOTE
  Missing or unexported object: ‘tidyselect::eval_select’
```
cause it's a new function...